### PR TITLE
Isaicp 2371

### PR DESCRIPTION
**Pull request by sandervd**

### CODE REVIEW

File: web/modules/custom/rdf_entity/src/RouteProcessor/RouteProcessorRdf.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * Constructs a new RouteProcessorCurrent.
+   *
```
Comment (pfrenssen) [inline web/modules/custom/rdf_entity/src/RouteProcessor/RouteProcessorRdf.php]: ... a new RouteProcessorRdf object.

File: web/modules/custom/rdf_entity/src/RouteProcessor/RouteProcessorRdf.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   * {@inheritdoc}
+   */
+  public function processOutbound($route_name, Route $route, array &$parameters, BubbleableMetadata $bubbleable_metadata = NULL) {
+    if ($route->hasOption('parameters')) {
+      foreach ($route->getOption('parameters') as $type => $parameter) {
+        // If the rdf_entity converter exists in the parameter,
+        // then the parameter is of type rdf_entity and needs to be normalized.
+        if (isset($parameter['converter']) && $parameter['converter'] == 'paramconverter.rdf_entity') {
+          $parameters[$type] = str_replace('/', '\\', $parameters[$type]);
+        }
```
Comment (pfrenssen) [inline web/modules/custom/rdf_entity/src/RouteProcessor/RouteProcessorRdf.php]: Wow, just wow. We can actually see who is responsible for making changes to the URLs and only act on them. This is miles ahead of how this worked in D7!

File: web/modules/custom/collection/collection.behat.inc [hunk 3]
```diff
@@ -351,6 +352,17 @@ class CollectionSubContext extends DrupalSubContextBase implements DrupalSubCont
       $memberships = Og::membershipStorage()->loadMultiple($membership_ids);
       Og::membershipStorage()->delete($memberships);
 
+      // Some more workarounds for og_menu.
+      $storage = \Drupal::getContainer()->get('entity.manager')->getStorage('ogmenu_instance');
+      $values = [
+        OgGroupAudienceHelper::DEFAULT_FIELD => $collection->id(),
+      ];
```
Comment (pfrenssen) [inline web/modules/custom/collection/collection.behat.inc]: I suppose this workaround should also be fixed when OG cleans up its orphans. I guess the menus are also group content?

Comment (idimopoulos) [inline web/modules/custom/collection/collection.behat.inc]: Well yes I hope so.
Menus are group content, yes.

File: web/modules/custom/collection/src/Form/JoinCollectionForm.php [hunk 4]
```diff
@@ -53,7 +53,7 @@ public function buildForm(array $form, FormStateInterface $form_state, AccountPr
         '#type' => 'link',
         '#title' => $this->t('Leave this collection'),
         '#url' => Url::fromRoute('collection.leave_confirm_form', [
-          'rdf_entity' => $collection->sanitizedId(),
```
Comment (idimopoulos) [inline web/modules/custom/collection/src/Form/JoinCollectionForm.php]: We got rid of it :) yay :)

**Pull request merged**